lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -1349,8 +1366,9 @@ joinStakePool ctx knownPools getPoolStatus apiPoolId (ApiT wid) body = do
         (txMeta, txTime)
         #pendingSince
   where
+    -- Not forecasting into the future. Should be safe.
     ti :: TimeInterpreter IO
-    ti = timeInterpreter (ctx ^. networkLayer @t)
+    ti = unsafeRunExceptT . timeInterpreter (ctx ^. networkLayer @t)
```

Comment:
It doesn't seem optimal to have to argue "should be safe". But I'm not sure that is easy to solve.

Reply:
`unsafe` will throw and cause a 500, we shouldn't use it if that can indeed happen. What about instead having a proper error, even if it's just a server error, but at least not an opaque 500.

Reply:
> instead having a proper error

Should at least use a custom error message instead of `unsafeRunExceptT`, and have the motivation why it is impossible _there_ instead of in a comment.